pkg/apis/v2v/v1alpha1/virtualmachineimport_types.go

```diff
@@ -83,36 +80,65 @@ type VirtualMachineImportStatus struct {
-	Conditions               []VirtualMachineImportCondition `json:"conditions"`
-	State                    VirtualMachineImportState       `json:"state"`
-	DataVolumes              []DataVolumeItem                `json:"dataVolumes"`
+	TargetVMName string                          `json:"targetVmName"`
+	Conditions   []VirtualMachineImportCondition `json:"conditions"`
+	DataVolumes  []DataVolumeItem                `json:"dataVolumes"`
 }
 
-// VirtualMachineImportState defines the state of virtual machine import
-// +k8s:openapi-gen=true
-type VirtualMachineImportState string
-
-// These are valid values of virtual machine import state
-const (
-	VirtualMachineImportStateRunning  VirtualMachineImportState = "Running"
-	VirtualMachineImportStateFinished VirtualMachineImportState = "Finished"
-	VirtualMachineImportStateUnknown  VirtualMachineImportState = "Unknown"
-)
-
 // VirtualMachineImportConditionType defines the condition of VM import
 // +k8s:openapi-gen=true
 type VirtualMachineImportConditionType string
 
 // These are valid conditions of of VM import.
 const (
-	// Ready represents status of the VM import process being completed.
-	Ready VirtualMachineImportConditionType = "Ready"
+	// Succeeded represents status of the VM import process being completed successfully
+	Succeeded VirtualMachineImportConditionType = "Succeeded"
 
 	// Processing represents status of the VM import process while in progress
 	Processing VirtualMachineImportConditionType = "Processing"
 )
 
+// SucceededConditionReason defines the reasons for the Succeeded condition of VM import
+// +k8s:openapi-gen=true
+type SucceededConditionReason string
+
+// These are valid reasons for the Succeeded conditions of VM import.
+const (
+	// ValidationFailed represents a failure to validate the eligibility of the VM for import
+	ValidationFailed SucceededConditionReason = "ValidationFailed"
+
+	// UpdatingSourceVMFailed represents a failure to stop VM or rename it on the source provider
+	UpdatingSourceVMFailed SucceededConditionReason = "ValidationFailed"
+
+	// VMCreationFailed represents a failure to create the VM entity
+	VMCreationFailed SucceededConditionReason = "VMCreationFailed"
+
+	// DataVolumeCreationFailed represents a failure to create data volumes based on source VM disks
+	DataVolumeCreationFailed SucceededConditionReason = "DataVolumeCreationFailed"
+
+	// VirtualMachineReady represents the completion of the vm import
+	VirtualMachineReady SucceededConditionReason = "VirtualMachineReady"
+)
+
+// ProcessingConditionReason defines the reasons for the Processing condition of VM import
+// +k8s:openapi-gen=true
+type ProcessingConditionReason string
+
+// These are valid reasons for the Processing conditions of VM import.
+const (
+	// Validating represents the validation of the mapping rules and eligibility of source VM for import
+	Validating ProcessingConditionReason = "Validating"
```

Comment:
Having these allowed reasons and condition types should the validation failure be represented as follows? 
- block: type: "Processing", reason: "Validating", status: "False"
- warn:  type: "Processing", reason: "Validating", status: "True"

Reply:
This validation reason includes any possible cause for validation failures:
* Secret not found
* Mappings resource not found
* Failure to connect to source provider
* Failure to to fetch VM details
* Failure to prepare mapping rules
* Violation of mapping rules

Therefore I think that validation is entitled for its own condition in order to provide a meaningful feedback about the cause of the failure above.

Would you see a value if separating the validating step from being tracked by the 'Processing' condition ?
And we'll have two conditions for monitoring the progress:
'Validating' - verifies the eligibility of the input for the import process.
'Processing' - once the inputs are verified, monitors the processing of the vm import process

Then 'Validating' condition will a similar behavior as you suggested:
* block: type: "Validating", reason: "MappingRulesViolated", status: "False"

But in order to be able and record the warning, we should add a dedicated condition for it, else, it will be overridden by next update to that condition.

You should add a condition 'MappingCheck' with reasons:
- MappingRulesViolated - status 'False' // message should contain the violations 
- MappingRulesHaveWarnings - status 'True' // message should contain the warnings 
- MappingRulesReady - status 'True'

Feel free to propose other naming.

Reply:
Yes, I think we should have separate condition types for reporting validation problems.

Reply:
I think type "Validating" and reasons you proposed look good.